pallets/phala/src/fat_tokenomic.rs

```diff
@@ -58,10 +59,12 @@ pub mod pallet {
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
 		ContractDepositChanged {
+			cluster: Option<ContractClusterId>,
```

Comment:
When can it be None?

Reply:
Users can stake to any address. If it is not a registered contract address, it would be None.

Reply:
Right. But when it happens, no cluster will receive the deposit change. As a result, the stake will be meaningless?

Reply:
Yes.
They can adjust the stake later when the contract becomes active or retrieve the stake back if it was a mistake.
I don't reject the meaningless stake to keep the logic as simple as possible.
If we reject them, we need to consider whether a user staked to an installed contract and the contract gets removed someday, then the user might get into trouble to retrieve his money back. 
We need to implement some extra logic to handle that case which would make the code complex.

Reply:
I don't think we will ever allow deleting the contracts. I don't have a preference now. We can stick to the current implementation or add the cluster check.

In the future I think we will move staking to the cluster, implemented with a driver contract.

Reply:
> I don't think we will ever allow deleting the contracts.

Even clusters can be destroyed. Not sure what strategies would be taken in the future.
Anyway, I would like to keep it simple if there aren't hard requirements. This would make the system safer and more robust and let us make further decisions with more confidence.